### PR TITLE
Feature/commons 44 setenv fixture

### LINKED ISSUE #44: Scalatest: Environment variable testing fixture
Testing code that depends on environment variables is challenging as there is the standard Java API doesn't allow for modifying env variables at runtime. In tests however, we sometimes need to set certain environment variables before the test and clean them up after. It could be done by implementing a fixture. E.g.

```scala
class MySpec ... with EnvFixture {
  it should "set FOO variable for this test body only" in {
    setEnv("FOO", 42)
    // execute some test code that reads FOO environment variable
    // via the standard Java API like System.getenv("FOO")
  }
}
```

### PR DESCRIPTION
**Pull request by wajda**
fixes #44

### CODE REVIEW

**Review by yruslan — DISMISSED** [review]
Very useful

File: src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.scalatest.{BeforeAndAfter, Suite}
+import za.co.absa.commons.reflect.ReflectionUtils
+
+import scala.collection.JavaConverters._
+
+trait EnvFixture extends BeforeAndAfter {
+  this: Suite =>
+
+  private val m = ReflectionUtils.extractFieldValue[ju.Map[String, String]](System.getenv, "m")
```
Comment (dk1844) [inline src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: It seems to me, that this map `m` is what is expected to be found via `System.getenv`, but there is no `System.setX` involved, the content of `m` never actually gets to the environment properties.

Therefore, at least currently, the test suite does not work
https://github.com/AbsaOSS/commons/pull/45/files#diff-be115859ff93a482acdc11a331867bc9c0eadf80fde0762bf2907acd0010b15aR28-R33

Comment (wajda) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: It works on the CI and my local. What java version are you testing it on? 
The `m` is an object reference to the underlying map that holds all the environment variables. It isn't supposed to be modified from Java API, that's why I used reflection.

Comment (wajda) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: > so the content of m never actually gets to the environment properties

It's not meant to be propagated to the real system environment, of cause. The goal is to fool `System.getenv()` for testing purposes.

Comment (dk1844) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: I am using
```OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_242-b08)```

It really fails for me, that why I assumed that you just missed something there:
```
EnvFixtureSpec:
- should set environment variable *** FAILED ***
  null did not equal "42" (EnvFixtureSpec.scala:31)
```

Comment (wajda) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: That's even more interesting.
```
openjdk version "1.8.0_275"
OpenJDK Runtime Environment (build 1.8.0_275-b01)
OpenJDK 64-Bit Server VM (build 25.275-b01, mixed mode)
```
image

Comment (wajda) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: Ah, you are using `AdoptOpenJDK`. It probably makes difference

Comment (wajda) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: @dk1844 I have tested it in 8 different JDKs and the test is passed everywhere:
- Open JKD - 1.8, 11
- Adopt (both hotspot and openj9) - 1.8, 11
- Amazon corretto - 1.8, 11

All were the latest public patch versions, including (AdoptOpenJDK)(build 1.8.0_275-b01)

Comment (yruslan) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: Works for me, both in IDE and `mvn test`
```
openjdk version "1.8.0_222"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_222-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.222-b10, mixed mode)
```

Comment (yruslan) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: After updating JDK, still works for me
```
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)
```

Comment (cerveada) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: On Windows different collection is used for `getenv(String)` and `getenv()`

```
    static String getenv(String name) {
        return theCaseInsensitiveEnvironment.get(name);
    }

    static Map<String,String> getenv() {
        return theUnmodifiableEnvironment;
    }
```

Comment (wajda) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: Case insensitivity :facepalm:

Comment (dk1844) [reply src/main/scala/za/co/absa/commons/scalatest/EnvFixture.scala:29]: Nice find, @cerveada!

Having updated to `OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_275-b01)`, of course there is no change in behavior.

**Comment by dk1844** [comment]
Now it works for me on Windows on maven level (`mvn clean && mvn test` succeeds). However, running the tests individually in IDEA fails.

**Review by dk1844 — APPROVED** [review]

**Review by yruslan — APPROVED** [review]

**Pull request merged**